simulator/interface/ISimulator.ts

```diff
@@ -1,11 +1,12 @@
 ﻿import { IOptions } from "./IOptions";
 import { IParseResult } from "../../parser/interface/IParseResult";
 import { IState } from "./IState";
+import { IRoundResult } from "./IRoundResult";
 
 export interface ISimulator {
 
     initialise(options: IOptions, warriors: IParseResult[]): void;
-    run(): void;
-    step(steps?: number): boolean;
+    run(): IRoundResult;
```

Comment:
I'll have to check how I'm handling this UI side I presume I'll need some tweaks to handle the results from these areas.

Reply:
My assumption (which may be wrong) is that you weren't really using the return value anyway, you were relying on the corresponding `ROUND_END` message to be published...